rust/sealevel/client/src/core.rs

```diff
@@ -15,9 +19,33 @@ use crate::{
 use hyperlane_core::H256;
 use hyperlane_sealevel_igp::accounts::{SOL_DECIMALS, TOKEN_EXCHANGE_RATE_SCALE};
 
+pub(crate) fn adjust_gas_price_if_needed(chain_name: &str, ctx: &mut Context) {
```

Comment:
not a huge deal but because we so many txs, you could maybe put it in `Context::new`

Reply:
I don't think we have a `chain` or `local_domain` available when we call `Context::new`